### ISSUE TITLE
🔃 🐞 Bug Report: Color-scheme of Elementor Editor v3.12 affects the front-end - [ED-10299]

### ISSUE THREAD

**Issue by typometre** [elementor/elementor#21809]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Updating to v3.12 of Elementor, The background of my website change from white to black.
If I set Elementor Editor in light color scheme, it doesn't affect the website but with Detection or Dark its change my front end.
I am on mac OS with Chrome with the Dark theme set in mac OS and retreived by Chrome. 
Was thinking that Elementor color scheme should only affect the editor and not the front even if not background color are set or only set to transparent.

### Steps to reproduce

On mac OS set the mode to dark mode on Desktop Settings.
On Elementor 3.12 set the mode to Dark or Detection
The front-end inspector show that it load the file theme-dark.min.css

`css media="(prefers-color-scheme: dark)"
:root {
    color-scheme: dark; … `

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.54 (Unix)
	MySQL version: Source distribution v10.3.32
	PHP Version: 8.0.23
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 64M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: https://starter.givememore.net
	Home URL: https://starter.givememore.net
	WP Multisite: No
	Max Upload Size: 64 Mo
	Memory limit: 256M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: fr-FR
	Timezone: Europe/Paris
	Admin Email: sites@givememore.fr
	Debug Mode: Inactive

== Theme ==
	Name: GMM-K
	Version: 1.0.0
	Author: Give me more
	Child Theme: Yes
	Parent Theme Name: Kadence
	Parent Theme Version: 1.1.35
	Parent Theme Author: Kadence WP

== User ==
	Role: administrator
	WP Profile lang: fr_FR
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Access Manager
		Version: 6.9.8
		Author: Vasyl Martyniuk <vasyl@vasyltech.com>

	Advanced Custom Fields: Extended
		Version: 0.8.9.3
		Author: ACF Extended

	Advanced Custom Fields PRO
		Version: 6.0.7
		Author: WP Engine

	Advanced Editor Tools (previously TinyMCE Advanced)
		Version: 5.6.0
		Author: Automattic

	BackupBuddy
		Version: 8.8.3
		Author: iThemes

	Complianz | GDPR/CCPA Cookie Consent
		Version: 6.4.3
		Author: Really Simple Plugins

	Disable Comments
		Version: 2.4.3
		Author: WPDeveloper

	Doubly Pro
		Version: 1.0.41
		Author: Unlimited Elements

	Dynamic.ooo - Dynamic Content for Elementor
		Version: 2.9.1
		Author: Dynamic.ooo

	Elementor
		Version: 3.12.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.12.0
		Author: Elementor.com

	Enable Media Replace
		Version: 4.1.0
		Author: ShortPixel

	Find My Blocks!
		Version: 3.5.4
		Author: Eddy Sims

	GMM Download List Block with Thumbnail
		Version: 2.0.4
		Author: Alexandre Dumas de Rauly

	GMM View Usage of Shortcodes
		Version: 1
		Author: Alexandre Dumas de Rauly

	Gutenberg
		Version: 15.4.0
		Author: Gutenberg Team

	Hide My WP Ghost
		Version: 7.0.11
		Author: WPPlugins

	iThemes Security Pro
		Version: 7.3.1
		Author: iThemes

	JC Submenu
		Version: 0.9.1
		Author: James Collings

	JSM's Show Post Metadata
		Version: 3.0.11
		Author: JS Morisset

	Kadence Blocks – Gutenberg Blocks for Page Builder Features
		Version: 3.0.27
		Author: Kadence WP

	Log cleaner for iThemes Security
		Version: 1.3.5
		Author: Michael Ott

	Matomo Analytics - Ethical Stats. Powerful Insights.
		Version: 4.14.1
		Author: Matomo

	Nested Pages
		Version: 3.2.3
		Author: Kyle Phillips

	Password Protected
		Version: 2.6.2
		Author: WPExperts

	Prevent Browser Caching
		Version: 2.3.4
		Author: Kostya Tereshchuk, ISRAEL IT

	Rank Math SEO
		Version: 1.0.110.2
		Author: Rank Math

	Regenerate Thumbnails Advanced
		Version: 2.4.1
		Author: ShortPixel

	SB Children block
		Version: 1.2.0
		Author: bobbingwide

	ShortPixel Image Optimizer
		Version: 5.2.1
		Author: ShortPixel - Convert WebP/AVIF & Optimize Images

	TinyMCE Custom Styles
		Version: 1.1.3
		Author: Tim Reeves (original author David Stöckl)

	Widget Detector for Elementor
		Version: 1.2.5
		Author: ElementInvader & FreelancersTools (Ivica Delić)

	WPB Accordion Menu or Category PRO
		Version: 2.2.8
		Author: wpbean

	WP Rocket
		Version: 3.12.6.1
		Author: WP Media

	WP SVG Images
		Version: 4.1
		Author: ShortPixel

	XO Featured Image Tools
		Version: 1.12.1
		Author: Xakuro


== Must-Use Plugins ==
	Sanitize File Name
		Version: 20180218
		Author: Mickaël Gris (Neticpro)


== Fonctionnalités ==
	Custom Fonts: 0
	Custom Icons: 0

== Intégrations ==
	


== Expériences d’Elementor ==
	Sortie DOM optimisée: Actif
	Amélioration du chargement des ressources: Actif
	Optimisation du chargement du CSS: Actif
	Icônes de police « inline »: Inactif par défaut
	Améliorations de l’accessibilité: Actif
	Points d’arrêt personnalisés supplémentaires: Actif
	admin_menu_rearrangement: Inactif par défaut
	Conteneur Flexbox: Actif
	Mettre à niveau la bibliothèque Swiper: Actif
	Par défaut vers le Nouveau constructeur de thème: Actif par défaut
	Editor Top Bar: Actif
	Page d’atterrissage: Actif par défaut
	Éléments imbriqués: Actif
	Chargement différé des images d’arrière-plan: Inactif par défaut
	Transitions des pages: Actif par défaut
	Notes: Actif par défaut
	Boucle: Actif
	Accroche de défilement: Actif par défaut
	Menu: Actif


== Journal ==
	
Log: showing 20 of 322023-01-18 14:27:53 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-01-18 14:27:57 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.1',
  'to' => '3.10.2',
)]
2023-01-18 14:27:59 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.1',
  'to' => '3.10.2',
)]
2023-01-19 14:13:05 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.5',
  'to' => '2.8.6',
)]
2023-01-26 15:27:41 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.6',
  'to' => '2.8.8',
)]
2023-01-26 15:27:42 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.6',
  'to' => '2.8.8',
)]
2023-02-09 13:25:37 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.8',
  'to' => '2.8.9',
)]
2023-02-09 13:26:09 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.1',
  'to' => '3.10.2',
)]
2023-02-09 13:26:38 [info] elementor::elementor_updater Started 
2023-02-09 13:26:41 [info] Elementor/Upgrades - _on_each_version Start  
2023-02-09 13:27:07 [info] Elementor/Upgrades - _on_each_version Finished 
2023-02-09 13:27:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.1',
  'to' => '3.10.2',
)]
2023-02-09 13:27:11 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.1',
  'to' => '3.10.2',
)]
2023-02-17 22:49:32 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.9',
  'to' => '2.8.10',
)]
2023-02-18 01:22:31 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.10',
  'to' => '2.8.11',
)]
2023-03-10 11:43:37 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.11',
  'to' => '2.8.12',
)]
2023-03-10 11:43:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.11',
  'to' => '2.8.12',
)]
2023-03-10 11:43:41 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.11',
  'to' => '2.8.12',
)]
2023-03-22 11:21:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.8.12',
  'to' => '2.9.0',
)]
2023-03-28 16:19:46 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.9.0',
  'to' => '2.9.1',
)]

JS: showing 7 of 7JS: 2023-01-18 13:37:23 [error X 14][https://starter.givememore.net/wp-admin/load-scripts.php?c=1&loadchunk_0=underscore,jquery-core,jquery-migrate,backbone,jquery-ui-core,jquery-ui-mouse,wp-api-request,regenerator-runtime,wp-polyfill,wp-&loadchunk_1=hooks&ver=6.1.1:4:51824] Cannot read properties of undefined (reading 'replace') 
JS: 2023-02-14 16:21:24 [error X 1][https://starter.givememore.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.0:3:791161] Cannot read properties of undefined (reading 'toLowerCase') 
JS: 2023-03-06 08:25:06 [error X 5][https://starter.givememore.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.2:3:907724] Cannot read properties of undefined (reading 'route') 
JS: 2023-03-13 15:18:34 [error X 1][https://starter.givememore.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.4:3:719506] Cannot read properties of undefined (reading 'widgets_settings') 
JS: 2023-03-13 19:30:22 [error X 1][https://starter.givememore.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.4:3:559769] Blocked a frame with origin "https://starter.givememore.net" from accessing a cross-origin frame. 
JS: 2023-03-14 00:05:03 [error X 14][https://starter.givememore.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.4:3:129011] C.getNamespaceArray is not a function 
JS: 2023-03-14 15:41:59 [error X 1][https://starter.givememore.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.5:3:748402] Cannot read properties of null (reading 'config') 

PHP: showing 3 of 3PHP: 2023-02-19 16:19:29 [warning X 14][/volume1/web/dev/starter/wp-content/plugins/elementor-pro/modules/posts/skins/skin-content-base.php::264] Attempt to read property "ID" on null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-06 09:46:11 [warning X 7][/volume1/web/dev/starter/wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-13 09:59:24 [error X 1][/volume1/web/dev/starter/wp-content/plugins/elementor/core/base/module.php::325] Uncaught Error: Class "ElementorPro\Modules\GlobalWidget\Widgets\Global_Widget" not found in /volume1/web/dev/starter/wp-content/plugins/elementor/core/base/module.php:325
Stack trace:
#0 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(308): Elementor\Core\Base\Module->init_widgets(Object(Elementor\Widgets_Manager))
#1 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array)
#2 /volume1/web/dev/starter/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/managers/widgets.php(122): do_action('elementor/widge...', Object(Elementor\Widgets_Manager))
#4 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/managers/widgets.php(327): Elementor\Widgets_Manager->init_widgets()
#5 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/elements/container.php(293): Elementor\Widgets_Manager->get_widget_types('theme-site-logo')
#6 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/base/element-base.php(1414): Elementor\Includes\Elements\Container->_get_default_child_type(Array)
#7 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/base/element-base.php(340): Elementor\Element_Base->get_child_type(Array)
#8 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/base/element-base.php(1459): Elementor\Element_Base->add_child(Array)
#9 /volume1/web/dev/starter/wp-content/plugins/elementor/includes/base/element-base.php(300): Elementor\Element_Base->init_children()
#10 /volume1/web/dev/starter/wp-content/plugins/elementor/core/dynamic-tags/dynamic-css.php(35): Elementor\Element_Base->get_children()
#11 /volume1/web/dev/starter/wp-content/plugins/elementor/core/dynamic-tags/dynamic-css.php(36): Elementor\Core\DynamicTags\Dynamic_CSS->render_styles(Object(Elementor\Includes\Elements\Container))
#12 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/post.php(178): Elementor\Core\DynamicTags\Dynamic_CSS->render_styles(Object(Elementor\Includes\Elements\Container))
#13 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/base.php(651): Elementor\Core\Files\CSS\Post->render_css()
#14 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/base.php(180): Elementor\Core\Files\CSS\Base->parse_content()
#15 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/base.php(130): Elementor\Core\Files\Base->update_file()
#16 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/base.php(217): Elementor\Core\Files\CSS\Base->update()
#17 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/post.php(201): Elementor\Core\Files\CSS\Base->enqueue()
#18 /volume1/web/dev/starter/wp-content/plugins/elementor/core/dynamic-tags/manager.php(488): Elementor\Core\Files\CSS\Post->enqueue()
#19 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(308): Elementor\Core\DynamicTags\Manager->after_enqueue_post_css(Object(Elementor\Core\DynamicTags\Dynamic_CSS))
#20 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#21 /volume1/web/dev/starter/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#22 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/base.php(264): do_action('elementor/css-f...', Object(Elementor\Core\Files\CSS\Post))
#23 /volume1/web/dev/starter/wp-content/plugins/elementor/core/files/css/post.php(201): Elementor\Core\Files\CSS\Base->enqueue()
#24 /volume1/web/dev/starter/wp-content/plugins/elementor-pro/modules/theme-builder/classes/locations-manager.php(93): Elementor\Core\Files\CSS\Post->enqueue()
#25 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(308): ElementorPro\Modules\ThemeBuilder\Classes\Locations_Manager->enqueue_styles('')
#26 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array)
#27 /volume1/web/dev/starter/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#28 /volume1/web/dev/starter/wp-includes/script-loader.php(2180): do_action('wp_enqueue_scri...')
#29 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(308): wp_enqueue_scripts('')
#30 /volume1/web/dev/starter/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array)
#31 /volume1/web/dev/starter/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#32 /volume1/web/dev/starter/wp-includes/general-template.php(3043): do_action('wp_head')
#33 /volume1/web/dev/starter/wp-content/themes/kadence/header.php(24): wp_head()
#34 /volume1/web/dev/starter/wp-includes/template.php(783): require_once('/volume1/web/de...')
#35 /volume1/web/dev/starter/wp-includes/template.php(718): load_template('/volume1/web/de...', true, Array)
#36 /volume1/web/dev/starter/wp-includes/general-template.php(48): locate_template(Array, true, true, Array)
#37 /volume1/web/dev/starter/wp-content/themes/kadence/single.php(14): get_header()
#38 /volume1/web/dev/starter/wp-includes/template-loader.php(106): include('/volume1/web/de...')
#39 /volume1/web/dev/starter/wp-blog-header.php(19): require_once('/volume1/web/de...')
#40 /volume1/web/dev/starter/index.php(17): require('/volume1/web/de...')
#41 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Dynamic.ooo - Dynamic Content for Elementor: Compatibilité non spécifiée
	Elementor Pro: Compatibilité non spécifiée
	Widget Detector for Elementor: Compatibilité non spécifiée

== Elementor Pro - Compatibility Tag ==
	

				

````

</details>

**Comment by m1ga** [comment]
Ah, that's why my font was suddenly white (browser is in dark mode). I fixed it by overriding it with my dark color again. But it was surprise to see the change just using a dark-mode default browser

**Comment by brandonbarringer** [comment]
+1 This affects all our sites even with the disable elementor's default colors checkbox checked.

**Comment by donwattz1459** [comment]
This is also effecting background colors of pages, form fields like input and select and the popup backgrounds.

**Comment by arielk** [comment]
Hi @typometre @m1ga @brandonbarringer @donwattz1459 

Does this also affect non-logged in users (visitors) or only logged in users with edit permissions?

**Comment by m1ga** [comment]
@arielk it looks like it's only when I'm logged in to the site (which is a bit of a relief). But since I was logged in while doing the update and was shocked that suddenly all my fonts turned white I've implemented a CSS fix already and didn't test it in incognito, cause time was ticking

**Comment by arielk** [comment]
Thanks @m1ga, so it has no effect on the users, it probably happened because we made an improvement to dark mode mechanism. We will release an update that removes this code from the front end soon. In the meantime you can change to light mode to deal with it.

**Comment by cenemil** [comment]
Hello @arielk, we also thought it is just affecting users with editing capabilities but it also happens on incognito.

**Comment by arielk** [comment]
@cenemil Can you please share with us a screenshot/cast or link so we can debug properly?

**Comment by cenemil** [comment]
@arielk unfortunately, I couldn't replicate it again as I have updated the UI settings of the sites that were affected while incognito. I'll let you know when I see another one today.

**Comment by corgalore** [comment]
We're seeing this behavior as well.

Examples of areas it's affecting:

- Popup background color.
- Input field background color (WooCommerce Qty).

When we roll back to the Light Mode for the editor, the affected background colors are no longer black, but become transparent, instead of white.

**Comment by eden-sonder** [comment]
Unfortunately this is affecting client sites that do not have the Hello theme installed. It is only doing it when logged in but is affecting pages that aren't utilizing Elementor. This has happened on the 3 client sites I have updated to the latest version of  Elementor, independently.

**Comment by Arian-c27** [comment]
You can't reproduce it with Hello theme because that theme has white background CSS for the body. The dark skin is applied to the root. Some themes do not add a default background color like that. Also, the background is not the only issue, any text that doesn't have color set on theme CSS or via elementor, inherits the root styles that are being loaded

**Comment by arielk** [comment]
This issue has been resolved in Elementor v3.12.1

Feel free to update

Thanks everyone!
